### PR TITLE
Move comm pkg to internal

### PR DESCRIPTION
**Pull request by mastersingh24**
#### Type of change
- Improvement (improvement to code structure)

#### Description
comm is not currently intended to be an external library.

### CODE REVIEW

**Review by C0rWin — DISMISSED** [review]

File: core/comm/alias.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+type CredentialSupport = pkgcomm.CredentialSupport
+type DynamicClientCredentials = pkgcomm.DynamicClientCredentials
+type GRPCClient = pkgcomm.GRPCClient
+type GRPCServer = pkgcomm.GRPCServer
+type KeepaliveOptions = pkgcomm.KeepaliveOptions
+type Metrics = pkgcomm.Metrics
+type SecureOptions = pkgcomm.SecureOptions
+type ServerConfig = pkgcomm.ServerConfig
+type ServerStatsHandler = pkgcomm.ServerStatsHandler
+type TLSOption = pkgcomm.TLSOption
```
Comment (sykesm) [inline core/comm/alias.go]: I assume there will be follow-ups to remove these? If not, the move doesn't really do much.

**Comment by cendhu** [comment]
@mastersingh24 @sykesm  A doubt but not related to this PR. If we have some util in the ledger and it is used only by the `core/ledger` (but different packages within the ledger), is it okay to move them to `internals/pkg/`. Is there a common guideline on what should go into `internals/pkg`? I understand that txflags used across different modules such as ledger, validation, etc... go here but I still wonder what to do with certain ledger utils.

File: core/comm/alias.go (hunk 1 above)
Comment (mastersingh24) [reply core/comm/alias.go]: > I assume there will be follow-ups to remove these? If not, the move doesn't really do much.

Yep.  Just did not want to change all the consumers at once.

**Comment by mastersingh24** [comment]
> @mastersingh24 @sykesm A doubt but not related to this PR. If we have some util in the ledger and it is used only by the `core/ledger` (but different packages within the ledger), is it okay to move them to `internals/pkg/`. Is there a common guideline on what should go into `internals/pkg`? I understand that txflags used across different modules such as ledger, validation, etc... go here but I still wonder what to do with certain ledger utils.

`internal/pkg` is intended to be used by internal packages which are intended to be used multiple different packages.

`internal` might be a better fit or perhaps an `internal` under `core/ledger` as well.

**Comment by cendhu** [comment]
Thanks, Gari. I will keep that in mind.

**Comment by mastersingh24** [comment]
ugh ... I hope these updates are not making the flakes worse

File: core/comm/alias.go (hunk 1 above)
Comment (sykesm) [reply core/comm/alias.go]: > Yep. Just did not want to change all the consumers at once.

Then my suggestion is to hold this out until the next piece is ready. Additional commits should be added until the type aliases are removed.

**Review by sykesm — CHANGES_REQUESTED** [review]
I'd prefer to see this deferred until the callers are updated. Type aliases are a good tool for compatibility but moving the code while preserving all of the aliases isn't really moving the ball forward.

If there's strong disagreement, I'll step aside but I do think we need to bring this to completion or not start it at all.

**Comment by mastersingh24** [comment]
> I'd prefer to see this deferred until the callers are updated. Type aliases are a good tool for compatibility but moving the code while preserving all of the aliases isn't really moving the ball forward.
> 
> If there's strong disagreement, I'll step aside but I do think we need to bring this to completion or not start it at all.

done ... I originally thought there might be more conflicts so was going to update one caller at a time.  But here it is in one shot

**Review by sykesm — APPROVED** [review]
Thank you.

**Pull request merged**